VTOL: move to cpp params API and some refactor

sfuhrer opened the pull request:
**Describe your solution**
Commit by commit:
- move VTOL module to px4::params
- set tailsitter flag through vehicle_status.is_tailsitter instead of VT_TYPE param, thus removing the dependency of some modules on vtol_type.h
- Commander/VTOL: move VT_FW_PERM_STAB param to commander and not pass it through as flag (less complexity, plus it actually can just as much be a fixed-wing param, not only VTOL)
- vtol_vehicle_status: replace several status bools by single vehicle_vtol_state --> it anyway can always only be in either MC, trans to FW, FW, trans to MC, so it makes it cleaner
- VTOL: remove some unsued variables (e.g. the mc weights were always set to 1 for the tailsitter, one can just as well remove them)
- Commander: remove permanent stabilization option for fixed-wing flight (alternative to https://github.com/PX4/PX4-Autopilot/pull/19412/commits/12a39491939c68b335992e9883476c05459bfe51)
- rename VT_PTCH_MIN to VT_PITCH_MIN and VT_LND_PTCH_MIN to Vt_LND_PITCH_MIN

**Test data / coverage**
SITL

**Additional context**

RomanBapst commented on ROMFS/px4fmu_common/init.d/airframes/13013_deltaquad:
```diff
@@ -31,6 +31,8 @@ param set-default BAT1_R_INTERNAL 0.0025
 param set-default CBRK_AIRSPD_CHK 162128
 param set-default CBRK_IO_SAFETY 22027
 
+param set-default COM_FW_PERM_STAB 1
```
@sfuhrer Need to check but if it's now possible to have stabilized mode mapped for multirotors then I suggest we get rid of this completely. In the past this was added because there was not stabilized mode for MC, just manual. So there was not way to fly manual MC, transition and end up in stabilized FW.

sfuhrer replied:
Very happy to remove it completely yes! I think in general it's a very bad idea to fly "Manual" in MC for VTOL, Stabilized it much clearer in that case. And if the user is flying in Manual FW and transitions to MC, then it's anyway clear that the system automatically enables stabilization (=implicit use of Stabilized mode, hopefully we can clean that up as well soonish).

sfuhrer replied:
I removed it.

RomanBapst commented:
@sfuhrer Awesome cleanup, which was overdue!
Looks really good to me.

RomanBapst commented:
@sfuhrer Can you check CI, there is an issue with the param VT_ELEV_MC_LOCK

sfuhrer commented:
> @sfuhrer Can you check CI, there is an issue with the param VT_ELEV_MC_LOCK

Should be addressed by https://github.com/PX4/PX4-Autopilot/pull/19412/commits/942d2c7f9a4866bfe633295e546b2c1378a04ee1

sfuhrer commented:
Flight tested on a Tiltrotor (Convergence):
https://review.px4.io/plot_app?log=e796fa50-ba72-4050-abcf-c615dbb88bf3
https://review.px4.io/plot_app?log=533a4729-b069-43df-a8c8-1713026b9b0a
No difference was noticeable in flight, though it seems like the RAM usage has increased. Will investigate.

sfuhrer commented on src/modules/vtol_att_control/vtol_type.h:
```diff
@@ -292,6 +254,44 @@ class VtolType
+					(ParamFloat<px4::params::VT_B_DEC_I>) _param_vt_b_dec_i,
+					(ParamFloat<px4::params::VT_B_DEC_MSS>) _param_vt_b_dec_mss,
+
+					(ParamFloat<px4::params::VT_PTCH_MIN>) _param_vt_ptch_min,
```
check if VT_PTCH_MIN or VT_PITCH_MIN

sfuhrer replied:
https://github.com/PX4/PX4-Autopilot/pull/19412/commits/5657b7b2cdf9d40224a093d8641001409bfddf34

sfuhrer commented:
@dagar CPU usage seems to increase by about 6% with this the first commit (https://github.com/PX4/PX4-Autopilot/pull/19412/commits/0c0631233df88ccf2cc04eebc18d1e08b883e4e4) of this PR (tested on a Pixhawk4 with the Convergence tiltrotor airframe).

current master:
```
timestamp: 356314641 (0.401839 seconds ago)
    load: 0.5133
    ram_usage: 0.6467
```
[ Moving of VTOL params to PX4::params ](https://github.com/PX4/PX4-Autopilot/pull/19412/commits/0c0631233df88ccf2cc04eebc18d1e08b883e4e4)  
```
cpuload
    timestamp: 5299963 (0.210583 seconds ago)
    load: 0.5714
    ram_usage: 0.6543
```
   Seems a bit much these 6%, do you have an idea what could be going on? Did I mess something up with using the param api with the present VTOL module inheritance structure?

sfuhrer commented:
With the CPU issues solved this is ready for proper review @RomanBapst @dagar .
Can also test fly it again.

RomanBapst commented on src/modules/commander/Commander.cpp:
```diff
@@ -2126,8 +2126,10 @@ Commander::run()
-				const bool is_rotary = is_rotary_wing(_status) || (is_vtol(_status) && _vtol_status.vtol_in_rw_mode);
-				const bool is_fixed = is_fixed_wing(_status) || (is_vtol(_status) && !_vtol_status.vtol_in_rw_mode);
+				const bool is_rotary = is_rotary_wing(_status) || (is_vtol(_status)
+						       && _vtol_status.vehicle_vtol_state != vtol_vehicle_status_s::VEHICLE_VTOL_STATE_FW);
```
```suggestion
						       && _vtol_status.vehicle_vtol_state == vtol_vehicle_status_s::VEHICLE_VTOL_STATE_MC);
```

sfuhrer replied:
No I think it has to be !=FW, as that also includes all the transition states (we want the type to be rotary in transition).

RomanBapst replied:
@sfuhrer Makes sense

RomanBapst commented on src/modules/vtol_att_control/vtol_att_control_main.cpp:
```diff
@@ -327,11 +328,21 @@ VtolAttitudeControl::Run()
+			// vehicle is doing a transition to FW
+			_vtol_vehicle_status.vehicle_vtol_state = vtol_vehicle_status_s::VEHICLE_VTOL_STATE_TRANSITION_TO_FW;
+
+			_fw_virtual_att_sp_sub.update(&_fw_virtual_att_sp);
```
@sfuhrer Why do you need to update here again, don't you update just before the switch?

sfuhrer replied:
No likely not, good find!

sfuhrer replied:
Removed it, see last commit

RomanBapst commented on src/modules/vtol_att_control/vtol_att_control_main.cpp:
```diff
@@ -327,11 +328,21 @@ VtolAttitudeControl::Run()
+			// vehicle is doing a transition to MC
+			_vtol_vehicle_status.vehicle_vtol_state = vtol_vehicle_status_s::VEHICLE_VTOL_STATE_TRANSITION_TO_MC;
 
 			_fw_virtual_att_sp_sub.update(&_fw_virtual_att_sp);
```
@sfuhrer Actually same here, do we need this?

sfuhrer replied:
No, see last commit

RomanBapst commented on src/lib/parameters/param_translation.cpp:
```diff
@@ -146,6 +146,25 @@ bool param_modify_on_import(bson_node_t node)
+	// 2022-04-11: translate VT_LND_PTCH_MIN to VT_LND_PITCH_MIN
+	{
+		if (strcmp("VT_LND_PTCH_MIN", node->name) == 0) {
+			strcpy(node->name, "VT_LND_PITCH_MIN");
```
@dagar What do we have as limit for parameter names nowdays. I remember it used to be 16 characters including the null termination?

dagar replied:
The limit is 16 characters not including NULL termination, but in some areas like mavlink you have to be careful because the storage might be char[16] (only terminated by NULL if the length is less than 16).

sfuhrer replied:
So I leave it at 16 I guess.

sfuhrer reviewed (commented):
Good catch Roman, I've removed the duplicated attitude setpoint updates. Please check last commit.

RomanBapst reviewed (dismissed):
@sfuhrer Looks good to me now.
@dagar Feel free to add your mustard as well.

junwoo091400 commented:
Maybe even move it to YAML parameter config in the future? Would it be beneficiary?

sfuhrer commented:
Thanks for the review @RomanBapst.
I've rebased and cleaned up the commits. 

>Maybe even move it to YAML parameter config in the future? Would it be beneficiary?

I'm not yet too accustomed to that way of defining params. What would be the benefit? Isn't it most useful when one has dynamic params (like in the allocation, where the amount of params is strongly dependent on the setup).
Anyway, my next plan was to revisit all the VTOL transition settings, that account for a large part of the VTOL params. I think we can clean up there quite a bit, and that's something both developers and users will appreciate.

sfuhrer commented on src/modules/vtol_att_control/tailsitter.cpp:
```diff
@@ -245,24 +234,24 @@ void Tailsitter::update_transition_state()
 	if (_vtol_schedule.flight_mode == vtol_mode::TRANSITION_FRONT_P1) {
 
-		const float trans_pitch_rate = M_PI_2_F / _params->front_trans_duration;
+		const float trans_pitch_rate = M_PI_2_F / _param_vt_f_trans_dur.get() ;
```
and increased min of param meta data in a93a4c54d00500a1ea2ca1e3608cca30be54480a
